datasets/ONS-International-exports-of-services-from-subnational-areas-of-the-UK/main.py

```diff
@@ -17,106 +17,279 @@
+            HDim(origin, 'Service Origin Geography',CLOSEST,LEFT),    
+]  
+tidy_sheet = ConversionSegment(tab, dimensions, observations)   
+savepreviewhtml(tidy_sheet, fname=tab.name + "Preview.html")
+trace.with_preview(tidy_sheet)
+#store 4b and appending to combined_dataframe
+trace.store("combined_dataframe", tidy_sheet.topandas())
 # -
 
-next_table['Export Services'].unique()
+#post processing
+df = trace.combine_and_trace(datasetTitle, "combined_dataframe")
+df
+
+df.rename(columns={'OBS' : 'Value','DATAMARKER' : 'Marker'}, inplace=True)
+
+#mapping Service Origin Geography outputs to nuts1 codes 
+trace.Service_Origin_Geography("mapping Service Origin Geography outputs to nuts1 nuts2 and nuts3 codes ")
+df['Service Origin Geography'] = df['Service Origin Geography'].map(
+    lambda x: {  
+'United Kingdom':'nuts1/all',
+'North East ':'nuts1/UKC',
+'North West':'nuts1/UKD',
+'Yorkshire and The Humber':'nuts1/UKE',
+'East Midlands':'nuts1/UKF',
+'West Midlands':'nuts1/UKG',
+'East of England':'nuts1/UKH',
+'London':'nuts1/UKI',
+'South East':'nuts1/UKJ',
+'South West':'nuts1/UKK',
+'Wales':'nuts1/UKL',
+'Scotland':'nuts1/UKM',
+'Northern Ireland':'nuts1/UKN'      
+        }.get(x, x))
+
+f1=(df['NUTS'] =='nuts2/')
+df.loc[f1,'Service Origin Geography'] = 'nuts2/' + df.loc[f1,'Service Origin Geography']
 
+f1=(df['NUTS'] =='nuts3/')
+df.loc[f1,'Service Origin Geography'] = 'nuts3/' + df.loc[f1,'Service Origin Geography']
 
+df['Period'] = "year/2017"
+trace.Period("Hard coded as year/2017")
+
+df['Export Services'] = df['Export Services'].apply(pathify)
+df
+
+df["Service Origin Geography"] = df["Service Origin Geography"].apply(pathify)
+
+df["Service Destination"] = df["Service Destination"].apply(pathify)
+
+df['Service Destination'] = df['Service Destination'].map(
+    lambda x: { 'total' : 'all', 'row' :'rest-of-world'}.get(x, x))
+
+df['Marker'] = df['Marker'].map(lambda x: { '..' : 'suppressed' }.get(x, x))
+
+df['Flow Directions'] = 'exports'
+trace.Flow_Directions("Adding Flow directions column to dataframe with value exports")
+
+#define the column order output into dataframe called tidy
+tidy = df[['Period','Export Services','Service Origin Geography','Service Destination', 'Flow Directions', 'Value','Marker']]
+tidy
+
+#output cube and spec
+cubes.add_cube(scraper, tidy.drop_duplicates(), "ONS-International-exports-of-services-from-subnational-areas-of-the-UK" )
+cubes.output_all()
+trace.render("spec_v1.html")
+
+# +
+#  Keeping for reference
+#  Unit = gbp-million
+#  , Measure Type = GBP Total
+#
+# Note the following do not have a NUTS code 
+# cambridgeshire-and-peterborough',
+# 'greater-manchester', 'liverpool-city-region', 'inner-london',
+# 'outer-london', 'north-of-tyne', 'sheffield-city-region',
+# 'tees-valley', 'west-of-england', 'cardiff-capital-region',
+# 'swansea-bay', 'aberdeen-and-aberdeenshire',
+# 'edinburgh-and-south-east-scotland', 'glasgow-city-region'
```

Comment:
I would maybe just made a note on the issue card for the DM picking this up regarding these areas did not have NUTS codes to represent them & was taken from the old code.

Reply:
Made a note on the card. Thank you.